android/app/src/main/java/com/coboltforge/dontmind/multivnc/AbstractScaling.java

```diff
@@ -3,116 +3,32 @@
  */
 package com.coboltforge.dontmind.multivnc;
 
-import android.widget.ImageView;
-
 /**
  * @author Michael A. MacDonald
  * 
  * A scaling mode for the VncCanvas; based on ImageView.ScaleType
  */
 abstract class AbstractScaling {
-	public static final int zoomableId = -1;
-	private static final int scaleModeIds[] = { zoomableId };
-	
-	private static AbstractScaling[] scalings;
-
-	static AbstractScaling getById(int id)
-	{
-		if ( scalings==null)
-		{
-			scalings=new AbstractScaling[scaleModeIds.length];
-		}
-		for ( int i=0; i<scaleModeIds.length; ++i)
-		{
-			if ( scaleModeIds[i]==id)
-			{
-				if ( scalings[i]==null)
-				{
-					switch ( id )
-					{
-				
-					case zoomableId:
-						scalings[i]=new ZoomScaling();
-						break;
-					}
-				}
-				return scalings[i];
-			}
-		}
-		throw new IllegalArgumentException("Unknown scaling id " + id);
-	}
-	
```

Comment:
These static fields & methods (along with `getByScaleType()` below) are used for creating new scaling instances. But we only have `ZoomScaling`. Now we create new `ZoomScaling`  object directly which makes all these obsolete.